File: app/src/main/java/com/example/challenge/base/ProfileViewModelFactory.java
package com.example.challenge.base;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.example.challenge.profile.ProfileViewModel;

public class ProfileViewModelFactory implements ViewModelProvider.Factory {
    private Application application;


    public ProfileViewModelFactory(Application application) {
        this.application = application;
    }


    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new ProfileViewModel(application);
    }
}

File: app/src/main/java/com/example/challenge/base/ProfileViewModelFactory.kt
package com.example.challenge.base

import android.app.Application
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModel
import com.example.challenge.profile.ProfileViewModel

class ProfileViewModelFactory(private val application: Application) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return ProfileViewModel(application) as T
    }
}